osf/quickfiles/models.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from django.core.exceptions import ValidationError
+
+
+from django.contrib.contenttypes.models import ContentType
+from osf.models import BaseFileNode, Folder, TrashedFileNode, OSFUser
+from osf.models.mixins import CleanMixin
+
+
+class QuickFileNode(BaseFileNode):
+    _provider = 'osfstorage'
+
+
+class QuickFolder(QuickFileNode, Folder, CleanMixin):
+    """
+    QuickFolder's a are specialized OsfStorageFolder models that are attached to the user instead of a
+    project, preprint, registration etc. QuickFolders have a few special restrictions normal Folder models don't have.
+
+    1. They must have a flat file structure. Quickfolders can have no parents nor can it's children be folders.
+    2. They must be attached to a user on creation, meaning the their target must always be a OSFUser instance.
+    3. If a user is deleted, disabled or marked as spam that user's quickfiles must become inaccessible.
+    4. Quickfiles must appear in ES results unless the user is deleted, disabled or marked as spam.
+    5. When loggable events (file uploads, renames .etc) happen to Quickfiles they are logged with that user's info,
+    though currently user logs aren't displayed.
+    6. A Quickfolder's title must reflect user's name in a grammatically correct way.
+    7. When a Quickfile is viewed/downloaded it only counts toward the metrics if the view/download is done by a user
+    who doesn't own the Quickfolder that file was in.
```

Comment:
I would normally disagree that we had to go all the way and create a QuickFolder class, but you've done a good job convincing me with these notes.